chia/plotting/manager.py

```diff
@@ -123,6 +189,8 @@ def _refresh_task(self):
             total_result: PlotRefreshResult = PlotRefreshResult()
             while self.needs_refresh() and self._refreshing_enabled:
                 batch_result: PlotRefreshResult = self.refresh_batch()
+                if batch_result.loaded_plots > 0 and self.needs_save_cache():
```

Comment:
it seems more reasonable to me to *always* save the cache when `loaded_plots > 0` (presumably meaning we found at least one new plot)

Reply:
I don't think that's a good idea, at least not in this place.. this could be done at the end but not after a batch.

Reply:
sure, hoist it out of the loop

Reply:
https://github.com/Chia-Network/chia-blockchain/pull/8009#issuecomment-909086002